lib/authenticate.js

```diff
@@ -26,96 +74,27 @@ var authenticate = function(client, username, password, options, callback) {
-      if (err) return handleCallback(callback, err, false);
-      _callback(null, true);
-    });
-  } else if (authMechanism === 'MONGODB-X509') {
-    client.topology.auth('x509', authdb, username, password, function(err) {
-      if (err) return handleCallback(callback, err, false);
-      _callback(null, true);
-    });
-  } else if (authMechanism === 'SCRAM-SHA-1') {
-    client.topology.auth('scram-sha-1', authdb, username, password, function(err) {
-      if (err) return handleCallback(callback, err, false);
-      _callback(null, true);
-    });
-  } else if (authMechanism === 'SCRAM-SHA-256') {
-    client.topology.auth('scram-sha-256', authdb, username, password, function(err) {
-      if (err) return handleCallback(callback, err, false);
-      _callback(null, true);
-    });
-  } else if (authMechanism === 'GSSAPI') {
-    if (process.platform === 'win32') {
-      client.topology.auth('sspi', authdb, username, password, options, function(err) {
-        if (err) return handleCallback(callback, err, false);
-        _callback(null, true);
-      });
-    } else {
-      client.topology.auth('gssapi', authdb, username, password, options, function(err) {
-        if (err) return handleCallback(callback, err, false);
-        _callback(null, true);
-      });
-    }
-  } else if (authMechanism === 'DEFAULT') {
-    client.topology.auth('default', authdb, username, password, function(err) {
-      if (err) return handleCallback(callback, err, false);
-      _callback(null, true);
-    });
-  } else {
-    handleCallback(
-      callback,
-      MongoError.create({
-        message: f('authentication mechanism %s not supported', options.authMechanism),
-        driver: true
-      })
-    );
-  }
-};
+  client.topology.auth(credentials, function(err) {
+    if (err) return handleCallback(callback, err, false);
+    _callback(null, true);
+  });
+}
 
-module.exports = function(self, username, password, options, callback) {
-  if (typeof options === 'function') (callback = options), (options = {});
-  options = options || {};
-
-  // Shallow copy the options
-  options = shallowClone(options);
-
-  // Set default mechanism
-  if (!options.authMechanism) {
-    options.authMechanism = 'DEFAULT';
-  } else if (
-    options.authMechanism !== 'GSSAPI' &&
-    options.authMechanism !== 'DEFAULT' &&
-    options.authMechanism !== 'MONGODB-CR' &&
-    options.authMechanism !== 'MONGODB-X509' &&
-    options.authMechanism !== 'SCRAM-SHA-1' &&
-    options.authMechanism !== 'SCRAM-SHA-256' &&
-    options.authMechanism !== 'PLAIN'
-  ) {
-    return handleCallback(
-      callback,
-      MongoError.create({
-        message:
-          'only DEFAULT, GSSAPI, PLAIN, MONGODB-X509, or SCRAM-SHA-1 is supported by authMechanism',
-        driver: true
-      })
-    );
-  }
+function authenticate(self, credentials, callback) {
+  // if (!VALID_AUTH_MECHANISMS.has(credentials.authMechanism)) {
```

Comment:
should this be commented out?

Reply:
I think I want to re-implement this, but it doesn't work right now.